File: api/src/Entity/Collectiontype.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Collectiontype
 *
 * @ORM\Table(name="cm_collectiontype", uniqueConstraints={@ORM\UniqueConstraint(name="collectionType_UN_type", columns={"type"})})
 * @ORM\Entity
 */
class Collectiontype
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private int $id;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=255, nullable=false)
     */
    private string $type;

    /**
     * @var bool|null
     *
     * @ORM\Column(name="active", type="boolean", nullable=true)
     */
    private ?bool $active;

    public function getId():int{
		return $this->id;
	}

	public function setId(int $id){
		$this->id = $id;
	}

	public function getType() : string{
		return $this->type;
	}

	public function setType(string $type){
		$this->type = $type;
	}

	public function getActive(): ?bool{
		return $this->active;
	}

	public function setActive(?bool $active){
		$this->active = $active;
	}


}

File: api/src/main/java/be/jimmyd/cm/entities/CollectionType.java
package be.jimmyd.cm.entities;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "cm_collectiontype")
public class CollectionType {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Basic
    @Column(name = "type")
    private String type;

    @Basic
    @Column(name = "active")
    private Boolean active;

    @OneToMany
    @JoinColumn(name = "typeid")
    private List<Collection> collections;

    @OneToMany
    @JoinColumn(name = "collectionbasetype")
    private List<Field> fields;

    private CollectionType(){

    }

    private CollectionType(Builder builder) {
        id = builder.id;
        type = builder.type;
        active = builder.active;
        collections = builder.collections;
        fields = builder.fields;
    }

    public int getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public Boolean getActive() {
        return active;
    }

    public List<Collection> getCollections() {
        return collections;
    }

    public List<Field> getFields() {
        return fields;
    }


    public static final class Builder {
        private int id;
        private String type;
        private Boolean active;
        private List<Collection> collections;
        private List<Field> fields;

        public Builder() {
        }

        public Builder withId(int val) {
            id = val;
            return this;
        }

        public Builder withType(String val) {
            type = val;
            return this;
        }

        public Builder withActive(Boolean val) {
            active = val;
            return this;
        }

        public Builder withCollections(List<Collection> val) {
            collections = val;
            return this;
        }

        public Builder withFields(List<Field> val) {
            fields = val;
            return this;
        }

        public CollectionType build() {
            return new CollectionType(this);
        }
    }
}
